### ISSUE TITLE
Scepticism about the correctness of the use of the LSTMCell

### ISSUE THREAD

**Issue by alirezakazemipour** [ikostrikov/pytorch-a3c#76]
Hi,

First of all, thank you very much for your easy-to-follow implementation!  Very intuitive and simple. :+1: 
My question is about your use of LSTMCell to implement the recurrent version of A3C.  
As it sounds from your code, you have used single timestep data (in your batch feeding to the model) to update your model including (the LSTMCell part) however, if it was intended to implement a recurrent procedure then you should have performed forward pass of the LSTMCell on each timestep in your recurrent part but it's missing thus, hidden states/cell states of the LSTMCell does not contribute to the gradient flow.
  
Simply put, I mean  such a part is missing:
```python
for i in inputs:
    # Step through the sequence one element at a time.
    # after each step, hidden contains the hidden state.
    out, hidden = lstm(i, hidden)
```
Otherwise, `hx` and `cx` **should not** have been detached:
https://github.com/ikostrikov/pytorch-a3c/blob/48d95844755e2c3e2c7e48bbd1a7141f7212b63f/train.py#L42
https://github.com/ikostrikov/pytorch-a3c/blob/48d95844755e2c3e2c7e48bbd1a7141f7212b63f/train.py#L43

Am I right? Or I'm missing something? 
Thank you in advance.
